pkg/functiontest/testing.go

```diff
@@ -0,0 +1,146 @@
+package functiontest
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"sigs.k8s.io/yaml"
+
+	"github.com/Azure/eno/pkg/function"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Scenario represents a test case for a synthesizer function.
+type Scenario[T function.Inputs] struct {
+	Name        string
+	Inputs      T
+	Assertion   Assertion[T]
+	ExpectError bool
+}
+
+// Evaluate runs the synthesizer function with the provided scenarios and asserts on the outputs.
+func Evaluate[T function.Inputs](t *testing.T, synth function.SynthFunc[T], scenarios ...Scenario[T]) {
+	for _, s := range scenarios {
+		t.Run(s.Name, func(t *testing.T) {
```

Comment:
can we add `t.Parallel()` by default? This is helpful for catching unexpected data race issues, although it's unlikely in this part

Reply:
good idea 👍